sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -83,15 +85,17 @@ case class Project(projectList: Seq[NamedExpression], child: LogicalPlan) extend
  */
 case class Generate(
     generator: Generator,
-    join: Boolean,
+    unrequiredChildOutput: Seq[Attribute],
```

Comment:
shall we add an assert to make sure `unrequiredChildOutput` are always resolved?

Reply:
not sure where, just add it to the "override lazy val resolved: Boolean"?

Reply:
we can just add it in the constructor.